jre_emul/Classes/com/google/j2objc/security/IosRSAKeyFactory.java

```diff
@@ -51,14 +51,18 @@ protected PublicKey engineGeneratePublic(KeySpec keySpec) throws InvalidKeySpecE
   @Override
   protected PrivateKey engineGeneratePrivate(KeySpec keySpec)
       throws InvalidKeySpecException {
-    if (keySpec instanceof X509EncodedKeySpec) {
+	  //The KeySpec for Private Key is PKCS8
+	 if (keySpec instanceof PKCS8EncodedKeySpec ) {
+	     return new IosRSAKey.IosRSAPrivateKey(((PKCS8EncodedKeySpec) keySpec).getEncoded());  
+	 }
+    if (keySpec instanceof X509EncodedKeySpec || keySpec instanceof PKCS8EncodedKeySpec ) {
       X509EncodedKeySpec x509Spec = (X509EncodedKeySpec) keySpec;
       return new IosRSAKey.IosRSAPrivateKey(x509Spec.getEncoded());
     } else if (keySpec instanceof RSAPrivateKeySpec) {
       return new IosRSAKey.IosRSAPrivateKey((RSAPrivateKeySpec) keySpec);
     }
     throw new InvalidKeySpecException(
-        "Must use RSAPublicKeySpec; was " + keySpec.getClass().getName());
+        "Must use PKCS8EncodedKeySpec; was " + keySpec.getClass().getName());
```

Comment:
Since this tests for PKCS8EncodedKeySpec, X509EncodedKeySpec and RSAPrivateKeySpec, change error message to "Invalid KeySpec type: " + keySpec.getClass...